src/utils/scrap_data.ts

```diff
@@ -0,0 +1,176 @@
+    url += `&${EXPLORE_API_PARAMS.price_range_max}=${wineParams.maxPrice}`;
+    if (wineParams.countryCodes) {
+      url += this.appendParam(wineParams.countryCodes, EXPLORE_API_PARAMS.country_codes);
+    }
+    if (wineParams.regions) {
+      url += this.appendParam(wineParams.regions, EXPLORE_API_PARAMS.region_ids, REGIONS);
+    }
+    if (wineParams.wineTypes) {
+      url += this.appendParam(wineParams.wineTypes, EXPLORE_API_PARAMS.wine_type_ids, WINE_TYPES);
+    }
+
+    return url;
+  };
+
+  mapWinesToWineObjects(response: any = []): Wine[] {
+    return response.map((wine: any) => {
+      return {
+        title: wine?.vintage?.name,
+        vintage: wine?.vintage?.year,
+        averagePrice: wine?.price?.amount,
+        currency: wine?.price?.currency?.code,
+        averageRating: wine?.vintage?.statistics?.ratings_average,
+        numberOfRatings: wine?.vintage?.statistics?.ratings_count,
+        imageUrl: wine?.vintage?.image?.location,
+        region: wine?.vintage?.wine?.region?.name,
+        country: wine?.vintage?.wine?.region?.country?.name,
+        bottleVolume: wine?.price?.bottle_type?.volume_ml,
+        discountPercent: wine?.price?.discount_percent,
+      };
+    });
+  }
+
+  getGrapes = async (page: Page): Promise<Grape[]> => {
+    const grapesResponse = await page.evaluate(async apiUrl => {
+      const cacheKey = localStorage.getItem('local_cache_key') + window['vivinoCacheKey'];
+      return fetch(`${apiUrl}?cache_key=${cacheKey}`).then((res: any) => {
+        return res.text().then((r: any) => JSON.parse(r));
+      });
+    }, VIVINO_GRAPES_API);
+    return <Grape[]>grapesResponse?.grapes;
+  };
+
+  isGrapeIsValid = (grapes: string): boolean => {
+    const grapesQuery = this.appendParam(grapes, EXPLORE_API_PARAMS.grape_ids, this.grapes);
+    if (grapesQuery.indexOf(EXPLORE_API_PARAMS.grape_ids) > -1) {
+      return true;
+    }
+    return false;
+  };
+
+  getWineStyles = async (page: Page): Promise<WineStyle[]> => {
+    const wineStylesResponse = await page.evaluate(async apiUrl => {
+      const cacheKey = localStorage.getItem('local_cache_key') + window['vivinoCacheKey'];
+      return fetch(`${apiUrl}?cache_key=${cacheKey}`).then((res: any) => {
+        return res.text().then((r: any) => JSON.parse(r));
+      });
+    }, VIVINO_WINE_STYLES_API);
+    return wineStylesResponse?.wine_styles;
+  };
+
+  getFoods = async (page: Page): Promise<Food[]> => {
+    const wineStylesResponse = await page.evaluate(async apiUrl => {
+      const cacheKey = localStorage.getItem('local_cache_key') + window['vivinoCacheKey'];
+      return fetch(`${apiUrl}?cache_key=${cacheKey}`).then((res: any) => {
+        return res.text().then((r: any) => JSON.parse(r));
+      });
+    }, VIVINO_FOODS_API);
+    return wineStylesResponse?.foods;
+  };
+
+  getAllWineListResponse = async (page: Page, wineParams: WineParams): Promise<Wine[]> => {
+    const winesList: Wine[] = [];
```

Comment:
In my opinion, naming is wrong.

Reply:
Okay, I gave the name like this because this function will fetch all APIs up to page N and return all the data.